TEDU-52 Revamped the Home page topics

swati-thacker13 opened the pull request:
- Based on feedback from the team on the Overview illustration, I created and modified the illustration.
- Revamped the Contributing topic and added more relevant sections to it.
- Revamped the Home page for Mautic documentation.
- Pulled out the Style guide and nested it under the Contributing section.
- Removed all the .rev files from the ‘Home’ folder.

RCheesley commented on pages/01.Home/02.Contributing/docs.en.md:
```diff
@@ -0,0 +1,321 @@
+Previously Gitbook was used to manage the Mautic documentation. However, this became cumbersome to maintain and difficult for those who were not familiar with Git. After exploring various solutions, the Education Team decided that Grav would best meet the needs of the Community.
+1. Go to the [the login page][mautic-docs-login]. Create a Mautic Community login, or log in with your existing forum/website account.
+1. Send a message on the [Documentation slack channel][mautic-docs-slack] to have your account raised to 'editor' - get a Slack invite [here][mautic-slack-invite].
+1. When logged in, click the **Edit page in Grav** link on the relevant page and make your changes.
```
We had to disable this feature for some reason I cannot recall, so I would suggest that folk use the UI

```suggestion
1. When logged in, go to the admin panel (ask the team for the link) and make your changes.
```

RCheesley commented on pages/01.Home/02.Contributing/docs.en.md:
```diff
@@ -0,0 +1,321 @@
+1. Select a file to edit on your fork.
+1. Make your commits.
+1. Open a pull request to `base fork: mautic/mautic-documentation` with `base: master`.
+1. Include and reference any [Issues][doc-issues] your Pull Request addresses.
```
```suggestion
1. Include and reference any [Issues][doc-issues] your Pull Request addresses. Be sure to write 'closes #0001' if your PR will close an existing issue, so that they are automatically linked.
```

RCheesley commented on pages/01.Home/02.Contributing/docs.en.md:
```diff
@@ -0,0 +1,321 @@
+
+#### Editing documentation using the command-line
+
+To edit documents using the git command-line system, read this section.
```
```suggestion
To edit documents using the Git command-line system, read this section.
```

RCheesley replied:
You may want to tell folk about the Github CLI utility - makes working at the command line much easier! https://cli.github.com/

swati-thacker13 replied:
I'll make a note about the GitHub CLI utility.

RCheesley commented on pages/01.Home/02.Contributing/docs.en.md:
```diff
@@ -0,0 +1,321 @@
+1. With the content of *README.md* visible, click on the pencil icon to begin editing the file.
+1. After you've made a change, scroll down to the *Commit changes* form. Saving your change requires describing what was changed and why.
+1. Before submitting your commit, select the box for *Creates a new branch* to start a new branch for your change. Name your branch something like `{yourusername}-revision-readme-file`
+1. Select *Propose file change*
```
You may want to add that if they are changing more than one file they can then select the newly created branch to switch to the branch, and then repeat this process until they have made all the edits they need, before making the PR.

swati-thacker13 replied:
Added this note:
If you are updating more than one file, then you can select the newly created branch to switch to the branch, and then repeat this process until you have made all the required edits, before creating a pull request.

RCheesley commented on pages/01.Home/02.Contributing/docs.en.md:
```diff
@@ -0,0 +1,321 @@
+[developer-docs-github]: <https://github.com/mautic/developer-documentation>
+
+[Mautic]: <https://mautic.org/>
+[mautic]: <https://mautic.org/>
```
We should not be duplicating link macros - let's pick one and update accordingly.

```suggestion
```

RCheesley commented on pages/01.Home/02.Contributing/01.Style Guide/docs.en.md:
```diff
@@ -0,0 +1,289 @@
+
+#### Relative links
+
+These will link to [`plugins/integration_test.html`][testing] on the documentation website created from the [_`plugins/integration_test.md`_][testing] source file.
```
This uses the old format, let's update it to avoid confusion!

```suggestion
These will link to [`/plugins/plugin-resources/testing-integrations`][testing] on the documentation website created from the [_`pages/13.Plugins/00.General%20Resources/01.Integration%20test/docs.en.md`_][testing] source file.  These should still use a link macro as with absolute links.
```

RCheesley commented on pages/01.Home/02.Contributing/01.Style Guide/docs.en.md:
````diff
@@ -0,0 +1,289 @@
+The first line is an example from within the same section.  The second for anywhere else in the directory structure.
+
+```markdown
+[testing]: <integration_test.html>
````
```suggestion
[testing]: </plugins/plugin-resources/testing-integrations>
```

RCheesley commented on pages/01.Home/02.Contributing/01.Style Guide/docs.en.md:
````diff
@@ -0,0 +1,289 @@
+```markdown
+[testing]: <integration_test.html>
+
+[testing]: <./../plugins/integration_test.html>
````
```suggestion
```
We do not need this any more, afaik

RCheesley commented on pages/01.Home/02.Contributing/01.Style Guide/docs.en.md:
```diff
@@ -0,0 +1,289 @@
+
+These will link to [`plugins/integration_test.html`][testing] on the documentation website created from the [_`plugins/integration_test.md`_][testing] source file.
+
+The first line is an example from within the same section.  The second for anywhere else in the directory structure.
```
```suggestion
```

We do not use that syntax any more

RCheesley commented on pages/01.Home/02.Contributing/01.Style Guide/docs.en.md:
```diff
@@ -0,0 +1,289 @@
+
+### Images
+
+Images should be placed in the media subfolders in the different sections of this repository.
```
```suggestion
Images should be placed in the same folder as the page they are used on. The image should be added to Git as a file to be included in the repository, not uploaded to Github and hosted by Github.
```

RCheesley commented on pages/01.Home/02.Contributing/01.Style Guide/docs.en.md:
```diff
@@ -0,0 +1,289 @@
+
+To display an image already in the documentation repository, use a relative path.
+
+The first line is an example of using images in the same section.  The second for anywhere else in the directory structure.
```
Please check if this is still this case - it was in our old Gitbook docs but I do not know if that is still the case now.

swati-thacker13 replied:
I tried using a relative link to link to an image somewhere else in the directory structure. It didn't work. So, removed this topic.

RCheesley commented on pages/01.Home/02.Contributing/01.Style Guide/docs.en.md:
```diff
@@ -0,0 +1,289 @@
+
+#### Absolute image links
+
+For images that cannot be uploaded via the GitHub web interface, upload them to a public URL service and use the generated link.
```
Do not want this to be happening - all images should be in the repo - let's remove this section (it often results in broken images and is subject to potential abuse)

swati-thacker13 replied:
Removed this topic.

RCheesley commented on pages/01.Home/02.Contributing/01.Style Guide/docs.en.md:
```diff
@@ -0,0 +1,289 @@
+
+> **Warning**
+>
+> ##### internal absolute image link
```
Remove this bit too, think it is not relevant.

swati-thacker13 replied:
Removed.

RCheesley reviewed (changes requested):
Fab effort - this will make a real difference for the newcomers!

Can you please also make sure that you remove the .revs folders as well as taking a look at the feedback?

Thanks, great work! 🙌  🚀

swati-thacker13 commented:
> Fab effort - this will make a real difference for the newcomers!
> 
> Can you please also make sure that you remove the .revs folders as well as taking a look at the feedback?
> 
> Thanks, great work! 🙌 🚀

swati-thacker13 commented on pages/01.Home/01.Overview of Mautic/docs.en.md:
```diff
@@ -0,0 +1,96 @@
+## Overview of Mautic
+---
+
+The Mautic platform helps you build a strong marketing strategy for your business. With Mautic, you can:
```
agree.

swati-thacker13 commented on pages/01.Home/01.Overview of Mautic/docs.en.md:
```diff
@@ -0,0 +1,96 @@
+
+- [Dashboard][dashboard]: A high-level representation of user data, Campaigns, and activities. You can create a highly personalized dashboard based on the information that you want to track.
+
+- [Reports][reports]: A tool to measure the effectiveness of your marketing Campaigns, Emails, Forms, or page hits. Reports also let you troubleshoot or workaround areas of improvement to enhance your marketing strategy.
```
@RCheesley, why capitalize 'Page'? It isn't a Mautic specific term.

RCheesley replied:
Page as in Landing Page - but maybe not needed in this context? Will leave it to your discretion! :)

swati-thacker13 commented on pages/01.Home/02.Contributing/docs.en.md:
```diff
@@ -0,0 +1,321 @@
+
+
+## Get Started
+ Although we support both Git and Grav for  documentation, we'd prefer contributions using Git because of the following reasons:
```
Changed 'Github' to 'GitHub'.

swati-thacker13 commented on pages/01.Home/01.Overview of Mautic/docs.en.md:
```diff
@@ -0,0 +1,96 @@
+
+Let us understand how the different pieces of Mautic fit together to give you a holistic marketing platform experience.
+
+![Mautic Overview](mautic-overview.png)
```
Replaced the puzzle piece with a white icon and updated the overview illustration.

swati-thacker13 commented:
Thanks for the appreciation, @RCheesley! I committed your suggestions and checked in a few more changes:

- Replaced the puzzle piece with a white icon and updated the overview illustration.
- Removed the topics from the style guide - **Relative image links** & **Absolute image links**.
- Added an introduction to the style guide.
- Added a new section called **Capitalization** to the style guide.
- Added a note about checking in multiple file changes with one pull request.

RCheesley reviewed (changes requested):
Great - one small issue with spacing which I could not see in the markdown:

![screenshot-docs ddev site-2021 02 27-21_40_06](https://user-images.githubusercontent.com/2930593/109401165-f4feac00-7944-11eb-83ae-38e419d7d196.png)

an extra p tag is sneaking in and the heading is misaligned.

swati-thacker13 commented:
@RCheesley, I aligned the 'Component' heading by removing the extra space. I don't see the extra <p> tag in markdown. Is this issue resolved now?

RCheesley reviewed (approved):
Awesome work, LGTM! 🚀
